userland/examples/solutions/sosp-ble-ess.c

```diff
@@ -58,14 +41,24 @@ int main(void)
 
   delay_ms(1500);
 
-  sensor_update_t *update = (sensor_update_t*) ipc_buf;
+  sensor_update_t *update = (sensor_update_t*) buf;
   ipc_register_client_cb(_svc_num, ipc_callback, update);
+  ipc_share(_svc_num, buf, 64);
 
-  update->type = SENSOR_HUMIDITY;
-  update->value = 185;
-  ipc_share(_svc_num, ipc_buf, 64);
+  int lux = ambient_light_read_intensity();
+  update->type = SENSOR_IRRADIANCE;
+  update->value = lux;
+  ipc_notify_svc(_svc_num);
 
+  int temp;
+  temperature_read_sync(&temp);
+  update->type = SENSOR_TEMPERATURE;
+  update->value = temp;
   ipc_notify_svc(_svc_num);
 
-  return 0;
+  unsigned humi;
+  humidity_read_sync(&humi);
+  update->type = SENSOR_HUMIDITY;
+  update->value = temp;
```

Comment:
This should be `update->value = humi;`?
Also, I'm not entirely sure how IPC works in Tock, but if the callback isn't pre-emptive then won't there only be three humidity updates? I think we might need an asynchronous update loop.

Reply:
Good call on both of these issues. I'm fixing them in the course-sosp branch right now